google/cloud/bigtable/admin_client_test.cc

```diff
@@ -23,6 +24,15 @@ namespace bigtable {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace {
 
+using ::testing::IsNull;
+using ::testing::NotNull;
+
+void CompareCQ(CompletionQueue const& a, CompletionQueue const& b) {
+  auto impl_a = google::cloud::internal::GetCompletionQueueImpl(a);
+  auto impl_b = google::cloud::internal::GetCompletionQueueImpl(b);
+  EXPECT_EQ(impl_a, impl_b);
```

Comment:
Note that all failures will point to this line of code, and it will be more difficult to troubleshoot problems.  There is a clever way to use matchers and move the `EXPECT_*()` to your test code.

Reply:
Well, ADL is complicated, but I would think yes.  You are also risking an ODR violation if we ever add `operator==` in the header file for that class 🤷